index.js

```diff
@@ -21,52 +50,119 @@ const SUBMISSION = [
 app.post('/signup', function(req, res) {
   // Add logic to decode body
   // body should have email and password
-
+  const {email,password,isAdmin } = req.body;
+    let present = false;
 
   //Store email and password (as is for now) in the USERS array above (only if the user with the given email doesnt exist)
-
+  
+      for(let i = 0;i<USERS.length;i++){
+           if(USERS[i].email === email){
+            present = true
+            res.send('username has already taken!')
+           }
+      }
+    
+    
+    if(!present){
+      const user1 = new User(email,password,isAdmin);
+      USERS.push(user1);
+      console.log(user1);
+      res.statusCode = 200;
+      res.send('Success..')
+    }
 
   // return back 200 status code to the client
-  res.send('Hello World!')
+ 
 })
 
+function generateRandomString(length) {
+  let result = '';
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  const charactersLength = characters.length;
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength));
```

Comment:
const randomIndex = Math.floor(Math.random() * charactersLength);
  result += characters.charAt(randomIndex % charactersLength);
  math.random may give a nu,ber greater than taken array , so better to use this method

Reply:
ok ,and thanks for your input

Reply:
changed it